### PR TITLE
Support globbing in sources parameter of salt.modules.archive (#32356)

### PR DESCRIPTION
**Pull request by mephi42**
### What does this PR do?
Adds globbing support to tar, zip and rar.

### What issues does this PR fix or reference?
32356

### Previous Behavior
Globbing symbol (*) is interpreted as is.

### New Behavior
Globbing symbol is expanded using glob.glob().

### Tests written?
No

### CODE REVIEW

File: tests/integration/modules/archive.py [hunk 1]
```diff
@@ -141,7 +141,7 @@ def test_tar_unpack(self):
 
         # Test extract archive
         ret = self.run_function('archive.tar', ['-xvf', self.arch], dest=self.dst)
-        self.assertTrue(isinstance(ret, list))
+        self.assertTrue(isinstance(ret, list), str(ret))
```
Comment (terminalmage) [inline tests/integration/modules/archive.py]: What is the purpose of this change?

**Review by terminalmage — COMMENTED** [review]
I would like to see this new behavior is documented in the functions' docstrings. I had a question too about the changes in the test suite.

File: tests/integration/modules/archive.py (hunk 1 above)
Comment (mephi42) [inline tests/integration/modules/archive.py]: When this check fails, it is hard to tell what exactly went wrong. This change makes test generate output like this:

```
======================================================================
FAIL: test_cmd_zip (integration.modules.archive.ArchiveTest)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "salt/tests/integration/modules/archive.py", line 188, in test_cmd_zip
    self.assertTrue(isinstance(ret, list), str(ret))
AssertionError: The minion function caused an exception: Traceback (most recent call last):
  File "salt/salt/minion.py", line 1479, in _thread_return
    return_data = executor.execute()
  File "salt/salt/executors/direct_call.py", line 28, in execute
    return self.func(*self.args, **self.kwargs)
  File "<string>", line 2, in cmd_zip
  File "salt/salt/utils/decorators/__init__.py", line 184, in wrapped
    return function(*args, **kwargs)
  File "salt/salt/modules/archive.py", line 653, in cmd_zip
    cmd.extend(_expand_sources(sources))
  File "salt/salt/modules/archive.py", line 422, in _expand_sources
    sources = path
UnboundLocalError: local variable 'path' referenced before assignment
```

and the problem becomes easier to debug.

Comment (terminalmage) [inline tests/integration/modules/archive.py]: Yeah, I was actually not familiar with the 2nd argument to ``asssertTrue()``, I looked at the unittest2 source to figure it out. Since I wasn't sure, I just wanted to make sure that it wasn't something where ``assertTrue()`` was consuming args using something like  ``*args`` in the function definition and asserting truth on all of them, because if that were the case then coercing it to a string would always result in a value that would evaluate as ``True``.

**Comment by terminalmage** [comment]
I opened https://github.com/mephi42/salt/pull/1 to get this new feature documented. Once that PR is merged into @mephi42's fork, we can get this in.

**Comment by mephi42** [comment]
Thank you for the documentation change, I have integrated it into my branch.

I made another change: in case filename does not contain special wildcard characters, it is returned as is. This is to avoid change in behavior, when missing files are silently dropped instead of being passed to archival tool, which should then fail with a meaningful error message.

**Comment by terminalmage** [comment]
Thanks!
